### PR TITLE
Add Cosine Annealing LR Scheduler

### PR DESCRIPTION
**Pull request by Kaixhin**
Adds the cosine annealing schedule from [SGDR: Stochastic Gradient Descent with Warm Restarts](https://arxiv.org/abs/1608.03983).

### CODE REVIEW

**Comment by soumith** [comment]
thanks Kai!

**Comment by Kaixhin** [comment]
[Docs came out weird](http://pytorch.org/docs/master/optim.html#torch.optim.lr_scheduler.CosineAnnealingLR), but I'm not sure why :/
